features/uploads/handlers.go

```diff
@@ -2,35 +2,49 @@ package uploads
 
 import (
 	"database/sql"
-	"github.com/S1ckret-Labs/family-archive-web-server/helpers"
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/s3"
-	"github.com/gin-gonic/gin"
-	"gopkg.in/guregu/null.v4"
 	"log"
 	"net/http"
 	"strconv"
 	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/gin-gonic/gin"
+	"gopkg.in/guregu/null.v4"
+
+	"github.com/S1ckret-Labs/family-archive-web-server/helpers"
 )
 
+// feature represents a feature with a database connection S3 | client and bucket name
 type Feature struct {
 	Db         *sql.DB
 	S3         *s3.S3
 	BucketName string
 }
 
+// CreateUploadRequest represents the request structure for creating an upload
 type CreateUploadRequest struct {
-	ObjectKey  string
-	SizeBytes  uint64
-	TakenAtSec null.Int
+	ObjectKey  string   `json:"objectKey"            example:"example_object_key"`
+	SizeBytes  uint64   `json:"sizeBytes"            example:"102400"`
+	TakenAtSec null.Int `json:"takenAtSec,omitempty"`
 }
 
+// CreateUploadRequestResult represents the result structure after creating an upload request
 type CreateUploadRequestResult struct {
-	ObjectId  uint64
-	ObjectKey string
-	UploadUrl string
+	ObjectId  uint64 `json:"objectId"  example:"123"`
+	ObjectKey string `json:"objectKey" example:"example_object_key"`
+	UploadUrl string `json:"uploadUrl" example:"https://s3.example.com/uploads/uploaded_file"`
 }
 
+// @Summary Get user's upload requests
+// @Description Get the upload requests for a user
+// @ID get-upload-requests
+// @Produce json
+// @Param id path uint64 true "User ID"
+// @Success 200
+// @Failure 400
+// @Failure 500
```

Comment:
Just curious. Can we also add description on why these error codes can happen?
e.g. `400 -> A user is not found in db`.

Reply:
yep, we can
I will do this in the future